examples_utils/benchmarks/environment_utils.py

```diff
@@ -68,6 +53,51 @@ def check_poprun_env_variables(benchmark_name: str, cmd: str):
         raise EnvironmentError(err)
 
 
+def enter_benchmark_dir(benchmark_dict: dict):
+    """Find and change to the path required to run the benchmark.
+
+    Notes:
+        For examples where the directory structure is non-standard (does not
+        follow the <category>/<model>/<framework>/'train.py' etc. convention),
+        the benchmark specification in the benchmarks.yml file will contain an
+        additional field 'location' which will inform this sub-module on how to
+        locate the python file that needs to be called.
+
+    Args:
+        benchmark_dict (dict): Dict created when evaluating the benchmark spec
+
+    """
+
+    # Find the root dir of the benchmarks.yml file
+    benchmark_path = Path(benchmark_dict["benchmark_path"]).parent
+
+    # If a special path is required, find and move to that in addition
+    if benchmark_dict.get("location"):
+        benchmark_path = benchmark_path.joinpath(benchmark_dict["location"])
+
+    os.chdir(benchmark_path)
+
+
+def get_mpinum(command: str) -> int:
```

Comment:
Can you obtain this information from popdist/poprun?

Reply:
This is in the case of mpirun, where poprun isnt used. 

Further, for apps that dont import popdist, we cant use popdist to query. Thought it simplest to take from the cmd which is explicit in its use of mpirun/poprun

Reply:
Unless popdist/this information is queryable from the framework itself?? but I suppose even then it would be a little more complicated to find out the framework and then make the check from the app